server/src/main/java/org/spine3/server/event/EventBus.java

```diff
@@ -375,7 +404,41 @@ public void close() throws Exception {
         private Builder() {
         }
 
+        /**
+         * Specifies an {@code StorageFactory} to configure this {@code EventBus}.
+         *
+         * <p>This {@code StorageFactory} instance will be used to create an instance of {@code EventStore}
+         * for this {@code EventBus}, <em>if</em> {@code EventStore} was not explicitly set in the builder.
+         *
+         * <p>Either a {@code StorageFactory} or an {@code EventStore} are mandatory to create an {@code EventBus}.
+         *
+         * @see #setEventStore(EventStore)
+         */
+        public Builder setStorageFactory(StorageFactory storageFactory) {
+            checkState(eventStore == null, EVENT_STORE_CONFIGURED_MESSAGE);
+            this.storageFactory = checkNotNull(storageFactory);
+            return this;
+        }
+
+        @Nullable
+        public StorageFactory getStorageFactory() {
+            return storageFactory;
+        }
+
+        /**
+         * Specifies {@code EventStore} to be used when creating new {@code EventBus}.
+         *
+         * <p>This method can be called if neither {@link #setEventStoreStreamExecutor(Executor)}
+         * nor {@link #setStorageFactory(StorageFactory)} were called before.
+         *
+         * <p>Either a {@code StorageFactory} or an {@code EventStore} must be set to create an {@code EventBus}.
+         *
+         * @see #setEventStoreStreamExecutor(Executor)
+         * @see #setStorageFactory(StorageFactory)
+         */
         public Builder setEventStore(EventStore eventStore) {
+            checkState(storageFactory == null, "storageFactory already set.");
```

Comment:
Codecov browser plugin shows that `null` branches are probably not tested. Please check.